test/Microsoft.Extensions.Configuration.UserSecrets.Test/ConfigurationExtensionTest.cs

```diff
@@ -43,6 +45,35 @@ private void SetSecret(string id, string key, string value)
         }
 
         [Fact]
+        public void AddUserSecrets_FindsAssemblyAttribute()
+        {
+            var configKey = nameof(AddUserSecrets_FindsAssemblyAttribute);
+            SetSecret("Microsoft.Extensions.Configuration.UserSecrets.Test", configKey, "true");
+            var userSecretId = Guid.NewGuid().ToString();
+            var config = new ConfigurationBuilder()
+                .AddUserSecrets(typeof(ConfigurationExtensionTest).GetTypeInfo().Assembly)
+                .Build();
+
+            Assert.Equal("true", config[configKey]);
+
+            SetSecret("Microsoft.Extensions.Configuration.UserSecrets.Test", configKey, "false");
```

Comment:
Why is this setting the value again? Is it trying to "undo" the change? Because if so, this isn't safe - the file could be left dirty if this test fails.